lib/java/com/google/android/material/slider/BaseSlider.java

```diff
@@ -472,7 +477,14 @@ private void processAttributes(Context context, AttributeSet attrs, int defStyle
             ? haloColor
             : AppCompatResources.getColorStateList(context, R.color.material_slider_halo_color));
 
-    tickVisible = a.getBoolean(R.styleable.Slider_tickVisible, true);
+    if (a.hasValue(R.styleable.Slider_tickVisibilityMode)) {
+      this.tickVisibilityMode = a.getInt(R.styleable.Slider_tickVisibilityMode, -1);
+    } else {
+      this.tickVisibilityMode = a.getBoolean(R.styleable.Slider_tickVisible, true)
```

Comment:
Declare a method to convert tickVisible to tickVisibilityMode to remove duplicate logic? Or just call setTickVisible() here.